test/e2e/aks_webhooks.go

```diff
@@ -0,0 +1,391 @@
+		ammp, err := getAzureManagedMachinePool(ctx, mgmtClient, mp)
+		Expect(err).NotTo(HaveOccurred())
+		By(fmt.Sprintf("Validating webhooks against AzureManagedMachinePool %s", ammp.Name))
+
+		By("Validating that we can't add a node label using the reserved 'kubernetes.azure.com/' prefix")
+		if ammp.Spec.NodeLabels == nil {
+			ammp.Spec.NodeLabels = map[string]string{}
+		}
+		ammp.Spec.NodeLabels[azureutil.AzureSystemNodeLabelPrefix+"/foo"] = "bar"
+		err = mgmtClient.Update(ctx, ammp)
+		Expect(err).To(HaveOccurred())
+		Expect(apierrors.IsInvalid(err)).To(BeTrue())
+
+		By("Validating that we can't update AzureManagedMachinePool.Spec.OSType")
+		ammp, err = getAzureManagedMachinePool(ctx, mgmtClient, mp)
+		Expect(err).NotTo(HaveOccurred())
+		if to.String(ammp.Spec.OSType) == "Linux" {
+			ammp.Spec.OSType = to.StringPtr("Windows")
+		} else {
+			ammp.Spec.OSType = to.StringPtr("Linux")
+		}
+		err = mgmtClient.Update(ctx, ammp)
+		Expect(err).To(HaveOccurred())
+		Expect(apierrors.IsInvalid(err)).To(BeTrue())
+
+		By("Validating that we can't update AzureManagedMachinePool.Spec.SKU")
+		ammp, err = getAzureManagedMachinePool(ctx, mgmtClient, mp)
+		Expect(err).NotTo(HaveOccurred())
+		ammp.Spec.SKU = ammp.Spec.SKU + "_vInvalid"
+		err = mgmtClient.Update(ctx, ammp)
+		Expect(err).To(HaveOccurred())
+		Expect(apierrors.IsInvalid(err)).To(BeTrue())
+
+		By("Validating that we can't update AzureManagedMachinePool.Spec.OSDiskSizeGB")
+		ammp, err = getAzureManagedMachinePool(ctx, mgmtClient, mp)
+		Expect(err).NotTo(HaveOccurred())
+		ammp.Spec.OSDiskSizeGB = to.Int32Ptr(to.Int32(ammp.Spec.OSDiskSizeGB) + 1)
+		err = mgmtClient.Update(ctx, ammp)
+		Expect(err).To(HaveOccurred())
+		Expect(apierrors.IsInvalid(err)).To(BeTrue())
+
+		By("Validating that we can't add a new 'infrastructure.cluster.x-k8s.io/custom-header-' Annotation")
+		ammp, err = getAzureManagedMachinePool(ctx, mgmtClient, mp)
+		Expect(err).NotTo(HaveOccurred())
+		if ammp.ObjectMeta.Annotations == nil {
+			ammp.ObjectMeta.Annotations = map[string]string{}
+		}
+		ammp.ObjectMeta.Annotations[azure.CustomHeaderPrefix+"foo"] = "bar"
+		err = mgmtClient.Update(ctx, ammp)
+		Expect(err).To(HaveOccurred())
+		Expect(apierrors.IsInvalid(err)).To(BeTrue())
+
+		By("Validating that we can't add or remove from AzureManagedMachinePool.Spec.AvailabilityZones")
+		ammp, err = getAzureManagedMachinePool(ctx, mgmtClient, mp)
+		Expect(err).NotTo(HaveOccurred())
+		if ammp.Spec.AvailabilityZones == nil {
+			ammp.Spec.AvailabilityZones = []string{}
+		}
+		ammp.Spec.AvailabilityZones = append(ammp.Spec.AvailabilityZones, "99")
+		err = mgmtClient.Update(ctx, ammp)
+		Expect(err).To(HaveOccurred())
+		Expect(apierrors.IsInvalid(err)).To(BeTrue())
+
+		ammp, err = getAzureManagedMachinePool(ctx, mgmtClient, mp)
+		Expect(err).NotTo(HaveOccurred())
+		if len(ammp.Spec.AvailabilityZones) > 0 {
+			ammp.Spec.AvailabilityZones = ammp.Spec.AvailabilityZones[:len(ammp.Spec.AvailabilityZones)-1]
```

Comment:
If all we want to accomplish here is changing the slice, this would be a bit more concise:
```suggestion
			ammp.Spec.AvailabilityZones = ammp.Spec.AvailabilityZones[1:]
```